### PR TITLE
Lp 235 General Partners create page template make reachable ac1

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link

https://companieshouse.atlassian.net/browse/LP-235

### Change description

Added page template, the partnership name has a placeholder for now.
Added routing to be able to test, again with placeholders, the previous and next pages still need to be added. 

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: locales/en/translations.json [hunk 1]
```diff
@@ -45,6 +45,11 @@
         }
     },
 
+    "generalPartnerPage": {
+        "title": "You now need to tell us about the general partners",
+        "pageInfromation": "If the partnership has more than one general partner, you'll add them one at a time."
```
Comment (mattch1) [inline locales/en/translations.json:50]: Will sort this typo.

File: src/presentation/controller/registration/Routing.ts [hunk 2]
```diff
@@ -12,12 +12,15 @@ const START_URL = `${BASE_URL}/start`;
...
 export const NAME_TEMPLATE = RegistrationPageType.name;
 export const NEXT_TEMPLATE = RegistrationPageType.next;
+export const GENERAL_PARTNERS_TEMPLATE = RegistrationPageType.generalPartners;
 
 export const WHICH_TYPE_URL = `${BASE_URL}/${WHICH_TYPE_TEMPLATE}`;
 export const NAME_URL = `${BASE_URL}/${NAME_TEMPLATE}`;
 
 export const NEXT_URL = `${BASE_URL}/transaction/${TRANSACTION_ID}/submission/${SUBMISSION_ID}/${NEXT_TEMPLATE}`;
 
+export const GENERAL_PARTNERS_URL = `${BASE_URL}/${GENERAL_PARTNERS_TEMPLATE}`;
```
Comment (mattch1) [inline src/presentation/controller/registration/Routing.ts:22]: Not sure whether we need the transaction and submission id here.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/views/general-partners.njk [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{% extends "layout.njk" %}
+
+{% set pageName="Add general partner information" %}
```
Comment (mattch1) [inline src/views/general-partners.njk:3]: Might not need this.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/presentation/controller/registration/Routing.ts (hunk 2 above)
Comment (markpit) [reply src/presentation/controller/registration/Routing.ts:22]: yes we will need both id's in the url. Think of it as any page after the submission gets created in MongDB (name page?) will need the id's in the url so we know where to get and read data from.

Comment (lduranteau) [reply src/presentation/controller/registration/Routing.ts:22]: yes, need ids
and can you move it to line 19 to follow the journey order

File: src/presentation/controller/registration/Routing.ts [hunk 4]
```diff
@@ -39,10 +42,18 @@ const registrationRoutingNext = {
   pageType: RegistrationPageType.next,
 };
 
+const registrationRoutingGeneralPartners = {
```
Comment (lduranteau) [inline src/presentation/controller/registration/Routing.ts:45]: same follow journey order

File: src/presentation/controller/registration/Routing.ts [hunk 5]
```diff
@@ -39,10 +42,18 @@ const registrationRoutingNext = {
...
+  currentUrl: GENERAL_PARTNERS_URL,
+  nextUrl: NEXT_URL,
+  pageType: RegistrationPageType.generalPartners,
+};
+
 const list = [
   registrationRoutingPartnershipType,
   registrationRoutingName,
   registrationRoutingNext,
+  registrationRoutingGeneralPartners,
```
Comment (lduranteau) [inline src/presentation/controller/registration/Routing.ts:56]: same follow journey order

File: src/views/general-partners.njk [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+{% block beforeContent %}
+  {% include "includes/back-link.njk" %}
+{% endblock %}
+
+{% block content %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+ 
+        {% if (data['registrationOrTransition'] == "registration") %}
```
Comment (lduranteau) [inline src/views/general-partners.njk:11]: we don't have this data

I think Vernon said to remove this caption

File: src/presentation/controller/registration/Routing.ts (hunk 4 above)
Comment (mattch1) [reply src/presentation/controller/registration/Routing.ts:38]: Reordered.

File: src/presentation/controller/registration/Routing.ts (hunk 5 above)
Comment (mattch1) [reply src/presentation/controller/registration/Routing.ts:55]: Reordered.

File: src/views/general-partners.njk (hunk 6 above)
Comment (mattch1) [reply src/views/general-partners.njk]: Removed.

File: src/presentation/controller/registration/Routing.ts (hunk 2 above)
Comment (mattch1) [reply src/presentation/controller/registration/Routing.ts]: Added the ids in.

File: src/presentation/controller/registration/Routing.ts [hunk 7]
```diff
@@ -11,11 +11,14 @@ import PageType from "../PageType";
 const START_URL = `${BASE_URL}/start`;
 export const WHICH_TYPE_TEMPLATE = RegistrationPageType.whichType;
 export const NAME_TEMPLATE = RegistrationPageType.name;
+export const GENERAL_PARTNERS_TEMPLATE = RegistrationPageType.generalPartners;
 export const NEXT_TEMPLATE = RegistrationPageType.next;
 
 export const WHICH_TYPE_URL = `${BASE_URL}/${WHICH_TYPE_TEMPLATE}`;
 export const NAME_URL = `${BASE_URL}/${NAME_TEMPLATE}`;
 
+export const GENERAL_PARTNERS_URL = `${BASE_URL}/transaction/${TRANSACTION_ID}/submission/${SUBMISSION_ID}/${GENERAL_PARTNERS_TEMPLATE}`;
```
Comment (markpit) [inline src/presentation/controller/registration/Routing.ts:20]: we have a bit of duplication now with the url paths, can you make ${BASE_URL}/transaction/${TRANSACTION_ID}/submission/${SUBMISSION_ID}
into a BASE_WITH_IDS_URL ?

Comment (mattch1) [reply src/presentation/controller/registration/Routing.ts]: Promoted this to a constant.

**Comment by markpit** [comment]
LGTM

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by lduranteau — APPROVED** [review]

**Pull request merged**